### PR TITLE
update SDK and also fix api

### PR DESCRIPTION
**Pull request by gshriki**
1. update SDK and change all hooks
2. add DS as authorization bearer and handle in api endpoint

### CODE REVIEW

**Review by rishibhargava — APPROVED** [review]

File: api/data.js [hunk 1]
```diff
@@ -5,9 +5,8 @@ dotenv.config();
 
 export default async function handler(request, response) {
   const projectId = request.headers['x-project-id'] || process.env.REACT_APP_DESCOPE_PROJECT_ID;
-  console.log(`projectId=${projectId}`);
-  const cookies = request.cookies;
-  const session_token = cookies.DS; // extract from request. The value is stored typically in DS cookie.
+  const header = request.headers['authorization'];
+  const session_token = header.split(" ")[1]; // extract from request. The value is stored typically in DS cookie.
```
Comment (asafshen) [inline api/data.js:9]: I would show both ways (like we do in go sdk)
Cookie and headers 

Also worth being resilient to bad auth header

File: src/index.tsx [hunk 2]
```diff
@@ -19,13 +19,15 @@ declare global {
...
-	useEffect(() => {
-		if (authenticated) {
-		// when the app is loaded - get logged in user details
-		me();
-		}
-	}, [authenticated]);
+	const { isAuthenticated, isSessionLoading} = useSession()
+	// const { user } = useUser();
+
+	// useEffect(() => {
```
Comment (asafshen) [inline src/index.tsx:25]: Let's remove commented code

File: src/views/admin/dataTables/index.jsx [hunk 3]
```diff
@@ -21,14 +22,17 @@ export default function Settings() {
...
 	if (!data.loaded) {
 		// TODO - load data once, in useEffect
 		fetch("/api/data", {
 			method: "get",
 			headers: {
 				Accept: "application/json, text/plain, */*",
 				"Content-Type": "application/json",
-				"x-project-id": projectId
+				"x-project-id": projectId,
+				Authorization: 'Bearer '+sessionToken,
```
Comment (asafshen) [inline src/views/admin/dataTables/index.jsx:35]: Not critical, prefer string templates over concat

**Review by asafshen — COMMENTED** [review]
Great job 

I would also add the loading return values to this, can be added later ofc

File: src/views/admin/dataTables/index.jsx (hunk 3 above)
Comment (gshriki) [reply src/views/admin/dataTables/index.jsx:35]: agreed. will fix

File: src/index.tsx (hunk 2 above)
Comment (gshriki) [reply src/index.tsx:25]: done

File: api/data.js (hunk 1 above)
Comment (gshriki) [reply api/data.js:9]: will bring back cookie as comments.
what do you mean being resilient?

Comment (gshriki) [reply api/data.js:9]: like this?
`  const session_token = header?.split(" ")[1] ?? "" ;`

**Pull request merged**